osu.Game.Modes.Taiko/Objects/Drawable/Pieces/CentreHitCirclePiece.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+using OpenTK;
+
+namespace osu.Game.Modes.Taiko.Objects.Drawable.Pieces
+{
+    /// <summary>
+    /// A circle piece which is used to visualise CentreHit objects.
+    /// </summary>
+    public class CentreHitCirclePiece : CirclePiece
+    {
+        protected override Framework.Graphics.Drawable CreateIcon()

```

Comment:
just remove this and make `CirclePiece`'s `Content => innerLayer` ?

then you can just `Add()` these things in the load method. will clean up the weird null checks in `BashCirclePiece` too